osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -37,31 +37,33 @@ protected override double StrainValueOf(DifficultyHitObject current)
             var maniaCurrent = (ManiaDifficultyHitObject)current;
             double endTime = maniaCurrent.EndTime;
             int column = maniaCurrent.BaseObject.Column;
+            double closestEndTime = 100;
 
             double holdFactor = 1.0; // Factor to all additional strains in case something else is held
             double holdAddition = 0; // Addition to the current note in case it's a hold and has to be released awkwardly
 
             // Fill up the holdEndTimes array
             for (int i = 0; i < holdEndTimes.Length; ++i)
             {
-                // If there is at least one other overlapping end or note, then we get an addition, buuuuuut...
+                // If there is at least one other overlapping end or note, then we get an addition
                 if (Precision.DefinitelyBigger(holdEndTimes[i], maniaCurrent.StartTime, 1) && Precision.DefinitelyBigger(endTime, holdEndTimes[i], 1))
                     holdAddition = 1.0;
 
-                // ... this addition only is valid if there is _no_ other note with the same ending. Releasing multiple notes at the same time is just as easy as releasing 1
-                if (Precision.AlmostEquals(endTime, holdEndTimes[i], 1))
-                    holdAddition = 0;
-
                 // We give a slight bonus to everything if something is held meanwhile
                 if (Precision.DefinitelyBigger(holdEndTimes[i], endTime, 1))
                     holdFactor = 1.25;
 
+                closestEndTime = Math.Min(closestEndTime, Math.Abs(endTime - holdEndTimes[i]));
+
                 // Decay individual strains
                 individualStrains[i] = applyDecay(individualStrains[i], current.DeltaTime, individual_decay_base);
             }
 
             holdEndTimes[column] = endTime;
 
+            // The hold addition only is valid if there is _no_ other note with the same ending. Releasing multiple notes at the same time is just as easy as releasing 1
+            holdAddition *= 1 / (1 + Math.Exp(0.5 * (24 - closestEndTime)));
```

Comment:
Apologies if I misunderstood the context here. So with this we basically nullify any `holdAddition` to a hold whenever there is a "nearby" (in terms of timing, its 24ms which is about 1/12 under 190 bpm; I am not sure if anyone sets this time with any map specifically in mind but otherwise it does feel like a magic number here... but whatever) hold also released. That's fine. 

My bigger question is, imagine in a hypothetical 7k scenario let's say in a LN wall pattern we have holds at column 1, 2, 3, 4, 5, 6, 7 and we have a release at 5. Cool, the note gets its bonus as it should: releasing a hold within a LN wall is not easy. Now consider a different pattern: we have a release at column 5 and 6. Now due to this line both notes lose their bonus, while practically speaking the pattern is not any easier. 

I am not saying current calculation does not have the same problem. However since you are now nerfing a lot of these scenarios, I do think this changes will actually make some actually hard but underrated LN patterns even more underrated (let's say a pattern consisting two long holds on column 4/7 with streams of double/triple shorter holds on other columns. A lot of these shorter holds might lose their bonus).

I feel like maybe we should consider scaling `holdAddition` bonus to how many holds are currently being pressed, not making it a boolean solely rely on an arbitrary `closestEndtime` (since this literally means we only consider bonus by looking at the next holds, while potentially we can have 4~8 LNs going on at the same time... ).  I know it is probably beyond the scope of this PR, but given how much this change may affect rating for all LN patterns I think at least we should reconsider our approach

Reply:
See my review [here](https://github.com/ppy/osu/pull/17913#discussion_r855365900). It basically says the same thing, with @molneya addressing the concern.

Reply:
> I am not sure if anyone sets this time with any map specifically in mind but otherwise it does feel like a magic number here

The 24ms value was chosen as its about 3/4 of the 300G timing window, still leaving about 1/4 of the window to hit the notes perfectly as if you were playing them on the same timing. The exact number is slightly arbitrary, but this value gave a good balance between nerfing overrated patterns while still keeping the bonuses when they really are awkward.

> ... we have a release at column 5 and 6. Now due to this line both notes lose their bonus, while practically speaking the pattern is not any easier.

The original algorithm had this same problem. The purpose of this change is to make sure that no matter the ordering of the columns the end star rating will be the same. If multiple notes share the same release timing, you can think of it as them all sharing the same `holdAddition`, as it's applied because of column 1. Or in other words, a new `holdAddition` will be given for every new snap a hold end appears (when it meets the awkwardness conditions). Addressing issues with the `holdAddition` itself should probably be addressed in a different PR.